milli/src/update/index_documents/mod.rs

```diff
@@ -547,7 +547,7 @@ where
             pool.install(|| {
                 for k in crate::vector::arroy_db_range_for_embedder(embedder_index) {
                     let writer = arroy::Writer::new(vector_arroy, k, dimension);
-                    if writer.is_empty(wtxn)? {
+                    if !writer.need_build(wtxn)? && writer.is_empty(wtxn)? {
```

Comment:
hello, do we still need the `writer.is_empty` part? I would expect `need_build` to suffice?

Reply:
Huum I thought it was possible that you end up with a writer that doesn't need to be built even though the next one needs to be built?
To me, we are looking specifically for the last writer that doesn't need to be built AND that is not used by any document.
But I could be wrong. Let me know!

> it sounds strange to me to have no writer doesn't need build and then do it anyway...

```
                    if writer.need_build(wtxn)? {
                        writer.build(...)?;
                    } else if writer.is_empty(wtxn)? {
                        break;
                    }
```

Reply:
alright good catch. The revised version you suggest expresses the intention better